channel/src/main/java/com/synopsys/integration/alert/channel/jira2/cloud/delegate/JiraCloudIssueTransitioner.java

```diff
@@ -54,85 +33,27 @@ public JiraCloudIssueTransitioner(
     }
 
     @Override
-    protected void findAndPerformTransition(ExistingIssueDetails<String> existingIssueDetails, String transitionName) throws AlertException {
-        String issueKey = existingIssueDetails.getIssueKey();
-        logger.debug("Attempting the transition '{}' on the issue '{}'", transitionName, issueKey);
-        List<TransitionComponent> issueTransitions = retrieveTransitions(issueKey);
-        Optional<IdComponent> foundTransitionId = findTransitionIdByTransitionName(issueTransitions, transitionName);
-        if (foundTransitionId.isPresent()) {
-            performTransition(issueKey, foundTransitionId.get());
-        } else {
-            List<String> validTransitions = gatherValidTransitionNames(issueTransitions);
-            throw new IssueMissingTransitionException(issueKey, transitionName, validTransitions);
-        }
-    }
-
-    private List<TransitionComponent> retrieveTransitions(String issueKey) throws AlertException {
-        try {
-            TransitionsResponseModel transitionsResponse = issueService.getTransitions(issueKey);
-            return transitionsResponse.getTransitions();
-        } catch (IntegrationException e) {
-            throw new AlertException(String.format("Failed to retrieve transitions from Jira. Issue Key: %s", issueKey), e);
-        }
-    }
-
-    private Optional<IdComponent> findTransitionIdByTransitionName(List<TransitionComponent> transitions, String transitionName) {
-        return transitions
-                   .stream()
-                   .filter(transitionComp -> transitionComp.getName().equals(transitionName))
-                   .map(TransitionComponent::getId)
-                   .map(IdComponent::new)
-                   .findFirst();
+    protected StatusDetailsComponent getIssueStatus(String issueKey) throws IntegrationException {
+        return issueService.getStatus(issueKey);
     }
 
-    private List<String> gatherValidTransitionNames(List<TransitionComponent> transitions) {
-        return transitions
-                   .stream()
-                   .map(TransitionComponent::getName)
-                   .collect(Collectors.toList());
+    @Override
+    protected TransitionsResponseModel getIssueTransitions(String issueKey) throws IntegrationException {
+        return issueService.getTransitions(issueKey);
     }
 
-    private void performTransition(String issueKey, IdComponent transitionId) throws AlertException {
-        IssueRequestModel issueRequestModel = new IssueRequestModel(issueKey, transitionId, new IssueRequestModelFieldsBuilder(), Map.of(), List.of());
-        try {
-            issueService.transitionIssue(issueRequestModel);
-        } catch (IntegrationException e) {
-            throw new AlertException(String.format("Failed to transition issue in Jira. Issue Key: %s", issueKey), e);
-        }
+    @Override
+    protected void transitionIssue(IssueRequestModel issueRequestModel) throws IntegrationException {
```

Comment:
I'm thinking `executeTransitionRequest` or something with this one. We have a lot of abstracted methods that move toward this ultimate goal, so I think making it as explicit as possible will clarify the distinction between methods.